File: Go_Solutions/Solutions/projeuler/Problem006.go
package euler

import (
	"fmt"
	"time"
)

/*
Problem006 answers the problem at : https://projecteuler.net/problem=6
	* Problem 6:
		Find the difference between the sum of the squares of the first one hundred natural numbers and the square of the sum.
*/
func Problem006() {

	start := time.Now()
	ans := sumSquare(100) - squareSum(100)
	end := time.Now()

	fmt.Printf("\nAnswer to Problem 6 : %d\n", ans)
	fmt.Printf("Time Taken: %f seconds\n\n", end.Sub(start).Seconds())
}

func squareSum(n int) int {
	k := 0
	for i := 1; i <= n; i++ {
		k += i * i
	}
	return k
}

func sumSquare(n int) int {
	k := 0
	for i := 1; i <= n; i++ {
		k += i
	}
	return k * k
}

File: Python/Problem006.py
from time import time

start = time()
sqSum = lambda n: sum([i**2 for i in range(n+1)])
sumSq = lambda n: sum(range(n+1))**2

ans = sumSq(100) - sqSum(100)

if __name__ == '__main__':
    print(f"\nAnswer: { ans }")
    print(f"Time Taken: { time() - start }\n")